### PR TITLE
[backend-api] prevent forgotten file after install and uninstall backend api in monorepo + fix patch

### PR DESCRIPTION
**Pull request by pk16011990**
| Q             | A
| ------------- | ---
|Description, reason for the PR| command `phing backend-api-install backend-api-uninstall` created new file `project-base/config/packages/security.yml.orig` because patch contains small discrepancy. It can caused commiting this useles file.
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: packages/backend-api/install/install.sh [hunk 1]
```diff
@@ -62,20 +62,25 @@ function apply_patch () {
...
     local PATCH_REVERSED=`grep "Reversed" <<< ${PATCH_DRY_RUN}`
+    local PATCH_WARNING=`grep "Hunk #" <<< ${PATCH_DRY_RUN}`
     local PATCH_FAIL=`grep "FAILED" <<< ${PATCH_DRY_RUN}`
 
     if [ ! -z "${PATCH_REVERSED}" ]
     then
         echo "Applied patch detected: patch for ${FILE_PATH} already applied. Doing nothing"
+    elif [ ! -z "${PATCH_WARNING}" ]
+    then
+        echo "Patch was applied for ${FILE_PATH} however patching did not have perfect match! Are line numbers correct in patchfile?"
```
[the remark this answers is not in the document]
Comment (pk16011990) [reply packages/backend-api/install/install.sh]: I suggess `Patch can be applied for ${FILE_PATH} however...`?

In my opinion I think that warning will be missed during whole build (eg. build-demo-dev) and only warning does not guarantee preventing of this issue.

[the remark this answers is not in the document]
Comment (pk16011990) [reply packages/backend-api/install/install.sh]: or add removing of `.orig` files in this case

File: packages/backend-api/install/install.sh [hunk 2]
```diff
@@ -62,20 +62,25 @@ function apply_patch () {
...
+        AT_LEAST_ONE_PATCH_FAILED=1
```
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply packages/backend-api/install/install.sh:76]: > In my opinion I think that warning will be missed during whole build (eg. build-demo-dev) and only warning does not guarantee preventing of this issue.

Changed my mind. You're right about this.

File: packages/backend-api/install/install.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply packages/backend-api/install/install.sh:75]: I changed the line to 

```
Patch can be applied for config/packages/security.yml however patching did not have perfect match!
Verify patchfile and apply manually with 'patch -t project-base/config/packages/security.yml packages/backend-api/install/config/packages/security.yml.patch'
```

Is it ok for you, @pk16011990 ?

**Pull request merged**